apis/console/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+#![no_std]
+
+use core::fmt;
+use core::marker::PhantomData;
+use libtock_platform::allow_ro;
+use libtock_platform::allow_ro::AllowRo;
+use libtock_platform::share;
+use libtock_platform::subscribe::Subscribe;
+use libtock_platform::{ErrorCode, Syscalls};
+
+/// The console driver.
+///
+/// It allows libraries to pass strings to the kernel's console driver.
+///
+/// # Example
+/// ```ignore
+/// use libtock2::Console;
+///
+/// // Prints 0x45 and the app which called it.
+/// Console::print(0x45);
+/// ```
+pub struct Console<S: Syscalls>(S);
+
+impl<S: Syscalls> Console<S> {
```

Comment:
Would you be fine with me sending you a PR that you can review and merge into this PR? I should be able to get to it next week. I know the documentation for developing async APIs in `libtock-rs` is lacking, and I was planning to use the console API as an example.

Reply:
I'd be fine with it, but I'd rather see this merged without the async version, and later add the async as an improvement. This way I don't have to diverge from upstream while I add more work.

There are also no downsides to having only a sync version because there are no users of this anyway right now.

Reply:
If we merge this without an async API and I go back in and add an async API, I may change the sync API as well (to make it interact better with the async API). I'm not expecting it to be difficult for you to update your code to match. Is that alright with you?

If so, I would be okay with merging this without async APIs as long as we open an issue to add async APIs to this crate.